File: src/test/java/com/zepben/vertxutils/routing/handlers/params/QueryParamRuleTest.java
/*
 * Copyright 2020 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package com.zepben.vertxutils.routing.handlers.params;

import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.nullValue;

public class QueryParamRuleTest {

    private final RequestValueConverter<String, String> converter = RequestValueConverter.Companion.create("string", s -> s);

    @Test
    public void basicConstructor() {
        QueryParamRule<String> rule = QueryParamRule.Companion.of("test", converter);
        assertThat(rule.getName(), is("test"));
        assertThat(rule.getConverter(), is(converter));
        assertThat(rule.isRequired(), is(false));
        assertThat(rule.getDefaultValue(), is(nullValue()));
    }

    @Test
    public void defaultValueConstructor() {
        QueryParamRule<String> rule = QueryParamRule.Companion.of("test", converter, "default");
        assertThat(rule.getName(), is("test"));
        assertThat(rule.getConverter(), is(converter));
        assertThat(rule.isRequired(), is(false));
        assertThat(rule.getDefaultValue(), is("default"));
    }

    @Test
    public void isRequiredConstructor() {
        QueryParamRule<String> rule = QueryParamRule.Companion.ofRequired("test", converter);
        assertThat(rule.getName(), is("test"));
        assertThat(rule.getConverter(), is(converter));
        assertThat(rule.isRequired(), is(true));
        assertThat(rule.getDefaultValue(), is(nullValue()));
    }

}

File: src/test/java/com/zepben/vertxutils/routing/handlers/params/QueryParamRuleTest.kt
/*
 * Copyright 2026 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.zepben.vertxutils.routing.handlers.params

import com.zepben.testutils.junit.SystemLogExtension
import com.zepben.vertxutils.routing.handlers.params.RequestValueConverter.Companion.create
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.equalTo
import org.hamcrest.Matchers.nullValue
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.RegisterExtension

class QueryParamRuleTest {

    companion object {

        @JvmField
        @RegisterExtension
        val systemOut: SystemLogExtension = SystemLogExtension.SYSTEM_OUT.captureLog().muteOnSuccess()

    }

    private val converter = create("string") { s: String -> s }

    @Test
    fun basicConstructor() {
        val rule = QueryParamRule.of("test", converter)
        assertThat(rule.name, equalTo("test"))
        assertThat(rule.converter, equalTo(converter))
        assertThat(rule.isRequired, equalTo(false))
        assertThat(rule.defaultValue, nullValue())
    }

    @Test
    fun defaultValueConstructor() {
        val rule = QueryParamRule.of("test", converter, "default")
        assertThat(rule.name, equalTo("test"))
        assertThat(rule.converter, equalTo(converter))
        assertThat(rule.isRequired, equalTo(false))
        assertThat(rule.defaultValue, equalTo("default"))
    }

    @Test
    fun ofRequiredConstructor() {
        val rule = QueryParamRule.ofRequired("test", converter)
        assertThat(rule.name, equalTo("test"))
        assertThat(rule.converter, equalTo(converter))
        assertThat(rule.isRequired, equalTo(true))
        assertThat(rule.defaultValue, nullValue())
    }

}
